docs/examples/Classifying_Using_HMM.py

```diff
@@ -0,0 +1,232 @@
+# 'accuracy' score, essentially providing a measure of the similarity between two categorical
+# distributions.
+from stonesoup.hypothesiser.categorical import CategoricalHypothesiser
+
+hypothesiser = CategoricalHypothesiser(predictor=predictor, updater=updater)
+
+# %%
+# Data Associator
+# ---------------
+# We will use a standard :class:`~.GNNWith2DAssignment` data associator.
+from stonesoup.dataassociator.neighbour import GNNWith2DAssignment
+
+data_associator = GNNWith2DAssignment(hypothesiser)
+
+# %%
+# Prior
+# -----
+# As we are tracking in a categorical state space, we should initiate with a categorical state for
+# the prior. Equal probability is given to all 3 of the possible hidden classes that a target
+# might take (the category names are also provided here).
+from stonesoup.types.state import CategoricalState
+
+prior = CategoricalState([1 / 3, 1 / 3, 1 / 3], category_names=hidden_classes)
+
+# %%
+# Initiator
+# ---------
+# For each unassociated detection, a new track will be initiated. In this instance we use a
+# :class:`~.SimpleCategoricalInitiator`, which specifically handles categorical state priors.
+from stonesoup.initiator.categorical import SimpleCategoricalInitiator
+
+initiator = SimpleCategoricalInitiator(prior_state=prior, measurement_model=None)
+
+# %%
+# Deleter
+# -------
+# We can use a standard :class:`~.UpdateTimeStepsDeleter`.
+from stonesoup.deleter.time import UpdateTimeStepsDeleter
+
+deleter = UpdateTimeStepsDeleter(2)
+
+# %%
+# Tracker
+# -------
+# We can use a standard :class:`~.MultiTargetTracker`.
+from stonesoup.tracker.simple import MultiTargetTracker
+
+tracker = MultiTargetTracker(initiator, deleter, measurements, data_associator, updater)
+
+# %%
+# Tracking
+# ^^^^^^^^
+
+tracks = set()
+for time, ctracks in tracker:
+    tracks.update(ctracks)
+
+print(f"Number of tracks: {len(tracks)}")
+for track in tracks:
+    certainty = track.state_vector[np.argmax(track.state_vector)][0] * 100
+    print(f"id: {track.id} -- category: {track.category} -- certainty: {certainty}%")
+    for state in track:
+        meas_string = f"associated measurement: {state.hypothesis.measurement.category}"
+        print(f"{state.timestamp} -- {state.category} -- {meas_string}")
+    print()
+
+# %%
+# Metric
+# ^^^^^^
+# Determining tracking accuracy.
+# In calculating how many targets were classified correctly, only tracks with the highest
+# classification certainty are considered.
+
+excess_tracks = len(tracks) - len(ground_truths)  # target value = 0
+sorted_tracks = sorted(tracks,
+                       key=lambda track: track.state_vector[np.argmax(track.state_vector)][0],
+                       reverse=True)
+best_tracks = sorted_tracks[:3]
+true_classifications = {ground_truth.category for ground_truth in ground_truths}
+track_classifications = {track.category for track in best_tracks}
+
+num_correct_classifications = len(true_classifications & track_classifications)
```

Comment:
If a={1,2}, b={2,1}, len(a & b)=2, even though the elements do not match up in terms of their order.  I can see that you've reordered the tracks so they shouldn't necessarily be order-wise equal, but is there a way to rework this so we check that each object is correctly categorised, as opposed to just the total number?

Reply:
There's no determining what the correct ordering of the truths & tracks is. Since there are no kinematics to do a standard data association approach, we could do something along the lines of seeing which categorical distributions line up the best with each other, but in this instance I'm simply saying that we have the correct classification of target *somewhere*.